### DISCUSSION TITLE
Set keymap for nvim-cmp failed

### DISCUSSION THREAD

**Discussion by OriLiMu** [LazyVim/LazyVim#1859] [Q&A] [answered]
Here is my settings. mapping part doesnt work. How to fix? 

I check the config [here](https://github.com/hrsh7th/nvim-cmp#recommended-configuration) 

```lua
local cmp = require("cmp")

return {
  {
    "hrsh7th/nvim-cmp",
    dependencies = { "hrsh7th/cmp-emoji" },
    ---@param opts cmp.ConfigSchema
    opts = function(_, opts)
      table.insert(opts.sources, { name = "emoji" })
      table.insert(opts.sources, { name = "natdat" })
      table.insert(opts.sources, { name = "dictionary", keyword_length = 2 })
    end,
    mapping = cmp.mapping.preset.insert({
      ["<Tab>"] = cmp.mapping.confirm({ select = true }), -- Accept currently selected item. Set `select` to `false` to only confirm explicitly selected items.
    }),
  },
}

```

**Comment by dpetka2001** [comment, marked as answer]
There's an example about that in the [docs](https://www.lazyvim.org/configuration/examples). It's at the end of the page. I don't use it myself, but you can see how to make it work.

**Reply by OriLiMu** [reply]
It worked. But I know the issue now. 
The <Tab> keymap in my case not work when you just enter nvim. 
But after a file save it(means type `:w<CR>`) works fine. 

So I do some test, I add <tab> and <C-t> with same function. 

Here is the case: 
Case 1:
When I enter nvim 
1. use <Tab> to select first -> not work
2. Then I use <C-t> -> work.
3. use <Tab> again -> work. 

Case 2:
When I enter nvim 
1. use <Tab> to select first -> not work
2. Save file. 
3. use <Tab> again -> work. 

Case 3: 
When I enter nvim 
1. use <C-t> to select first -> work
2. use <Tab>  -> work. 


This means save file or use other keymap with same will fix this issue, I think there may some conflict 
with tab keymap. 
How to fix it? 



Here is my code: 
```lua
return {
  {
    "hrsh7th/nvim-cmp",
    dependencies = { "hrsh7th/cmp-emoji" },
    ---@param opts cmp.ConfigSchema
    opts = function(_, opts)
      table.insert(opts.sources, { name = "emoji" })
      table.insert(opts.sources, { name = "natdat" })
      table.insert(opts.sources, { name = "dictionary", keyword_length = 2 })
      local cmp = require("cmp")
      opts.mapping = vim.tbl_extend("force", opts.mapping, {
        ["<Tab>"] = cmp.mapping.confirm({ select = true }), -- Accept currently selected item. Set `select` to `false` to only confirm explicitly selected items.
        ["<C-t>"] = cmp.mapping.confirm({ select = true }), -- Accept currently selected item. Set `select` to `false` to only confirm explicitly selected items.
      })
    end,
  },
}
```

This issue may not that serious I can live on with it. May Solve it later

**Comment by ogios** [comment]
set it in opts. (i use tab and s-tab to select)

```lua
  {
    "hrsh7th/nvim-cmp",
    opts = function(_, opts)
      local cmp = require("cmp")
      opts.mapping = vim.tbl_extend("force", opts.mapping, {
        ["<Tab>"] = cmp.mapping.select_next_item({ behavior = cmp.SelectBehavior.Insert }),
        ["<S-Tab>"] = cmp.mapping.select_prev_item({ behavior = cmp.SelectBehavior.Insert }),
      })
    end,
  },
```

**Reply by iamshm** [reply]
I tried this, but it didnt work on my end.
